src/main/kotlin/solve/importer/ProjectParser.kt

```diff
@@ -110,7 +114,61 @@ object ProjectParser {
             )
         }
         landmarks.forEach { (name, outputs) ->
-            frames.add(ProjectFrame(name, Path(images[name]!!), outputs))
+            try {
+                val longName = name.toLong()
+                frames.add(ProjectFrame(longName, Path(images[name]!!), outputs))
+            } catch (e: Exception) {
+                frames.add(
+                    ProjectFrame(
+                        0,
+                        Path(images[name]!!),
+                        outputs,
+                        errorMsg = mutableListOf("$name can not converted to Long"),
+                        errorType = mutableListOf(ErrorType.INCORRECT_IMAGE_NAME)
+                    )
+                )
+            }
+        }
+        frames.forEach {
+            it.landmarkFiles.forEach { a ->
+                when (a.projectLayer.kind) {
+                    LayerKind.Plane ->
+                        try {
+                            ImagePlanesParser.parse(a.path.toString())
```

Comment:
On the future: can there be not only one error during parsing? What if, for example, there are not unique uids and invalid positions togegher in the file?

Reply:
If sequence is big, this can take a lot of time. This should be done asynchronously with loading indicator.